pilot/pkg/xds/ads.go

```diff
@@ -182,45 +182,45 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-	if !s.preProcessRequest(con.proxy, req) {
+	if !s.shouldProcessRequest(con.proxy, req) {
 		return nil
 	}
 
-	// For now, don't let xDS piggyback debug requests start watchers
-	if strings.HasPrefix(req.TypeUrl, "istio.io/debug") {
+	// For now, don't let xDS piggyback debug requests start watchers.
+	if req.TypeUrl == v3.DebugType {
```

Comment:
I think the prefix part is intended

Reply:
is there a case where there is stuff after `istio.io/debug`?

Reply:
Ah. Ok. Will add it